### ISSUE TITLE
add option to return original path instead of a cached copy

### ISSUE THREAD

**Issue by nicoechaniz** [kineapps/flutter_file_dialog#12]
For our project: https://gitlab.com/elRepo.io

which is a distributed file sharing app based on RetroShare, we need to be able to share existing files on the client instead of copies of them. 
Would it be possible to adapt your plugin to include this option? -> return the original file path instead of making a copy. We would use this in conjunction with localOnly: true.

It is an important goal for our project to be able to share the files without using additional storage space and to share them from a non-volatile (not cache) storage because the system is P2P and it depends on the files not being deleted locally by the system when it needs space.

Please let us know if you think this is possible. We could collaborate with some humble funding to work on this feature request.

**Comment by kinex** [comment]
The returned path is a content provider path (not a file system path). What do you need the path for? For reading the file or displaying the path?

The function could be improved to return the full content provider URI and not only the path.  So instead of returning
**/document/primary:Download/image.jpg**

it could return
**content://com.android.externalstorage.documents/document/primary:Download/image.jpg**

For more information about content providers see:
https://developer.android.com/guide/topics/providers/content-provider-basics

See also:
https://stackoverflow.com/questions/5657411/android-getting-a-file-uri-from-a-content-uri

**Comment by nicoechaniz** [comment]
```quote
The returned path is a content provider path (not a file system path).
What do you need the path for? For reading the file or displaying the path?
```
We only need the path (the actual path to the existing file in the filesystem, not a new copy/cache) so we can tell the backgound service (Retroshare) to add the file to it's shared files. Retroshare will then read the file when it needs to. The rationale behind this is that the user must be able to share existing files without taking up additional space by copying them.

Is this possible?

**Comment by kinex** [comment]
This is now available in 2.1.0, see `OpenFileDialogParams.copyFileToCacheDir`

**Comment by selankon** [comment]
Hi, 

This return the filepath in android uri scheme, example:

```
content://com.android.providers.media.documents/document/image%3A2355
```
Is there a way to get absolute file path? thanks
